Use C99 types

JayKickliter opened the pull request:
`u_int32_t` is not a standard type in c, but it is defined on several platforms in `sys/types.h`. However, inert doesn't include that header. Instead, inert has relied on transitively-including it via `erl_driver.h` and `ei.h`. Something changed in OTP23, breaking that transitive include on some Linux distros. We might as well use `uint32_t` since inert already has a hard [dependency](https://github.com/msantos/inert/blob/master/c_src/Makefile#L29) on C99.

I ran include-what-you-use to make sure all the `#includes` are up to date:

```
$ bear rebar3 compile # generate compile_commands.json via build-ear
$ iwyu_tool.py -p compile_commands.json

inert_drv.c should add these lines:
#include <stdint.h>                 // for int32_t
#include <sys/_types/_u_int32_t.h>  // for u_int32_t
#include <sys/errno.h>              // for EBADF, EBUSY, EINVAL
#include <sys/fcntl.h>              // for fcntl, F_GETFD
#include "erl_drv_nif.h"            // for ErlDrvMonitor

inert_drv.c should remove these lines:
- #include <errno.h>  // lines 16-16
- #include <fcntl.h>  // lines 24-24
- #include <sys/time.h>  // lines 18-18
- #include <unistd.h>  // lines 23-23
- #include "ei.h"  // lines 27-27

The full include-list for inert_drv.c:
#include <stdint.h>                 // for int32_t
#include <stdio.h>                  // for NULL, size_t
#include <string.h>                 // for memset, strlen, memcpy
#include <sys/_types/_u_int32_t.h>  // for u_int32_t
#include <sys/errno.h>              // for EBADF, EBUSY, EINVAL
#include <sys/fcntl.h>              // for fcntl, F_GETFD
#include <sys/resource.h>           // for rlimit, getrlimit, RLIMIT_NOFILE
#include "erl_driver.h"             // for driver_select, ErlDrvData, driver...
#include "erl_drv_nif.h"            // for ErlDrvMonitor
---
```

msantos commented:
@JayKickliter thank you for fixing this! Sorry about the delay, going to have to check my notification settings, I didn't see this PR.

The patch for `u_int32_t` looks good. About the other header changes, the Linux documentation specifies for headers:
* [errno](https://man7.org/linux/man-pages/man3/errno.3.html)

~~~ C
#include <errno.h>
~~~

* [fcntl](https://man7.org/linux/man-pages/man2/fcntl64.2.html)

~~~ C
#include <unistd.h>
#include <fcntl.h>
~~~

* [getrlimit](https://man7.org/linux/man-pages/man2/getrlimit.2.html)

FreeBSD also requires `sys/types.h` while OpenBSD does not require `sys/time.h`.
~~~ C
#include <sys/time.h>
#include <sys/resource.h>
#include <sys/types.h>
~~~

* [erl_driver.h](https://github.com/erlang/otp/blob/master/erts/emulator/beam/erl_driver.h#L40)

  `erl_driver.h` includes `erl_drv_nif.h` so it shouldn't need to be added.

JayKickliter commented:
Thanks. I’ll just back out the header changes since everything else was working already.

JayKickliter committed:
Use C99 types

'u_int32_t' is not a standard type in c, but it is defined on several
platforms in 'sys/types.h'. However, inert doesn't include that
header. Instead, inert has relied on transitively-including it via
'erl_drriver.h' and 'ei.h'. Something changed in OTP23, breaking that
transitive include on some Linux distros.

JayKickliter commented:
Backed out. The only two changes now are u_int32_t -> uint32_t and removal of Solaris special-casing.
